src/resolvers/http-and-file.ts

```diff
@@ -1,14 +1,21 @@
 import { createResolveHttp, resolveFile } from '@stoplight/json-ref-readers';
 import { Resolver } from '@stoplight/json-ref-resolver';
 import { DEFAULT_REQUEST_OPTIONS } from '../request';
+import { Agent } from 'http';
 
-const resolveHttp = createResolveHttp(DEFAULT_REQUEST_OPTIONS);
+export interface IHttpAndFileResolverOptions {
+  agent?: Agent;
+}
 
 // resolves files, http and https $refs, and internal $refs
-export const httpAndFileResolver = new Resolver({
-  resolvers: {
-    https: { resolve: resolveHttp },
-    http: { resolve: resolveHttp },
-    file: { resolve: resolveFile },
-  },
-});
+export function createHttpAndFileResolver(opts?: IHttpAndFileResolverOptions): Resolver {
```

Comment:
Could we still export `httpAndFileResolver`? Otherwise this is likely to be a breaking for some.

```suggestion
export const httpAndFileResolver = createHttpAndFileResolver();

export function createHttpAndFileResolver(opts?: IHttpAndFileResolverOptions): Resolver {
```

Reply:
and doing so would let us avoid updating tests and code that rely on `httpAndFileResolver`

Reply:
Done.